### DISCUSSION TITLE
QuickAdd AI Assistant max tokens mapping to AI providers

### DISCUSSION THREAD

**Discussion by rmccorkl** [chhoumann/quickadd#724] [Q&A]
QQ need clarification

Under  the AI Assistant Settings  when configuring a provider   when you add a model your asked to input

- Model Name
- Max Tokens

**The question is on Max Tokens  does this map to the a) the Context window or b) the  Max Output ?**

Model | Description | Context window | Max output tokens | Training data
-- | -- | -- | -- | --
gpt-4o-mini | New GPT-4o-miniOur affordable and intelligent small model for fast, lightweight tasks. GPT-4o mini is cheaper and more capable than GPT-3.5 Turbo. Currently points to gpt-4o-mini-2024-07-18. | 128,000 tokens | 16,384 tokens

![Screenshot 2024-08-20 at 16 16 40](https://github.com/user-attachments/assets/a6b866f7-f5dc-4b41-af9f-f135f6352ecf)

**Comment by chhoumann** [comment]
Late answer, sorry - and a fair question, since the field does not say.

It is (a), the context window. QuickAdd uses Max Tokens to budget how much input it may send (for example when chunking long prompts) and to warn when an estimated prompt exceeds the configured context. So for gpt-4o-mini you would enter 128000, not 16384.

One nuance: for providers whose API requires an explicit output cap (Anthropic), QuickAdd derives a conservative output limit separately - this field is never sent as an output maximum on OpenAI-style APIs.

Also, on current versions you rarely need to type it by hand anymore: each provider has a model source / **Browse models** flow that imports models with this filled in automatically.

https://quickadd.obsidian.guide/docs/AIAssistant
